elasticsearch-transport/lib/elasticsearch/transport/transport/base.rb

```diff
@@ -42,7 +42,8 @@ def initialize(arguments={}, &block)
           @logger      = options[:logger]
           @tracer      = options[:tracer]
 
-          @sniffer     = options[:sniffer_class] ? options[:sniffer_class].new(self) : Sniffer.new(self)
+          sniffer_class = options[:sniffer_class] ? options[:sniffer_class] : Sniffer
+          @sniffer     = sniffer_class.new(self, logger)
```

Comment:
I'm changing it, but I'll point out that this completely removes the point of using a getter/setter. The point is that with the getter you can change the underlying logic. 

In the case of a logic that's pretty unlikely however.